mkosi/__init__.py

```diff
@@ -3203,11 +3205,28 @@ def nspawn_params_for_build_sources(args: MkosiArgs, sft: SourceFileTransfer) ->
     else:
         params += ["--chdir=/root"]
 
-    params += [f"--setenv={env}={value}" for env, value in args.environment.items()]
+    params += [
+        f"--setenv={env}={value}"
+        for env, value in extend_run_env(args, args.environment).items()
+    ]
 
     return params
 
 
+def extend_run_env(args: MkosiArgs, env: Mapping[str, str]) -> Mapping[str, str]:
```

Comment:
```suggestion
def extend_run_env(args: MkosiArgs, env: Mapping[str, str]) -> Dict[str, str]:
```
since it's clear what we return here. I think that will also make the cast further below unnecessary.

Reply:
Fixed. It seems you were right about the additional cast as well.